packages/zoe/README.md

```diff
@@ -56,17 +56,25 @@ agoric follow :published.priceFeed.ATOM-USD_price_feed
 
 A contract instance can be upgraded to use a new source code bundle in a process that is very similar to
 [upgrading a vat](https://github.com/Agoric/agoric-sdk/blob/master/packages/SwingSet/docs/vat-upgrade.md).
-Most state is discarded, however "durable" collections are retained for use by the replacement version.
 
 The upgrade process is triggered through the "adminFacet" of the instance, and requires specifying the new source code. (Note that a "null upgrade" that re-uses the original bundle is valid, and a legitimate approach to deleting accumulated state).
 
 ```js
 const results = E(instanceAdminFacet).upgradeContract(newBundleID);
 ```
 
-The new bundle itself must export a `prepare` function in place of `start`, and is obligated to redefine every durable Kind that was created by its predecessor.
+This will replace the behavior of the existing instnace with that defined in the bundle. The new behavior is an additional _incarnation_ of the instance. Most state is discarded, however "durable" collections are retained for use by the replacement version. 
 
-For example, suppose v1 code of a simple single-increment-counter contract anticipated extension of exported functionality  and decided to track it by means of "codeVersion" data in baggage. v2 code could add multi-increment behavior like so:
+There are a few requirements for the contract that differ from non-upgradable contracts:
+1. Export
```

Comment:
The section titles are a bit terse. I'd like to think of this as a checklist. Maybe:

* Entry point: export `prepare` function instead of `start`
* Durability: keep precious state in durable data, all other state is discarded
* Kinds: use `prepareExoClass` to re-attach all durable kinds
* Promptness: the `prepare` function must finish without awaiting messages from other vats

Reply:
Yeah that's what I had first and then I thought it better for people to not think they understand the requirements after reading the bullet points. I still think that. Is it persuasive?